File: src/main/java/me/deecaad/weaponmechanicsplus/weapon/modifiers/MeleeModifier.java
package me.deecaad.weaponmechanicsplus.weapon.modifiers;

import me.deecaad.core.file.SerializeData;
import me.deecaad.core.file.Serializer;
import me.deecaad.core.file.SerializerException;
import me.deecaad.weaponmechanicsplus.weapon.modifiers.util.ModifierHelper;
import org.jetbrains.annotations.NotNull;

public class MeleeModifier implements Serializer<MeleeModifier>, ModifierHelper {

    /**
     * Default constructor for serializer.
     */
    public MeleeModifier() {
    }


    @NotNull
    @Override
    public MeleeModifier serialize(SerializeData data) throws SerializerException {
        return null;
    }
}

File: src/main/kotlin/com/cjcrafter/weaponmechanicsplus/weapon/modifiers/MeleeModifier.kt
/*
 * Copyright (c) 2026. All rights reserved. Distribution of this file, similar
 * files, related files, or related projects is strictly controlled.
 */

package com.cjcrafter.weaponmechanicsplus.weapon.modifiers

import me.deecaad.core.file.*

class MeleeModifier : Serializer<MeleeModifier> {

    @Throws(SerializerException::class)
    override fun serialize(data: SerializeData): MeleeModifier {
        return MeleeModifier()
    }
}